cpp/src/arrow/memory_pool.cc

```diff
@@ -170,13 +170,8 @@ MemoryPoolBackend DefaultBackend() {
   if (backend.has_value()) {
     return backend.value();
   }
-#ifdef ARROW_JEMALLOC
-  return MemoryPoolBackend::Jemalloc;
-#elif defined(ARROW_MIMALLOC)
-  return MemoryPoolBackend::Mimalloc;
-#else
-  return MemoryPoolBackend::System;
-#endif
+  struct SupportedBackend default_backend = SupportedBackends().front();
+  return default_backend.backend;
```

Comment:
Or simply:
```c++
  return SupportedBackends().front().backend;
```

Reply:
Yup, I did test this locally to make sure, and it works like I expected it to: mimalloc is default with no `ARROW_DEFAULT_MEMORY_POOL` set and also with a `ARROW_DEFAULT_MEMORY_POOL` set that is not supported / non existent. 

Thanks!